### DISCUSSION TITLE
Watch mutation or state

### DISCUSSION THREAD

**Discussion by T0miii** [vuejs/pinia#1218] [Help and Questions] [answered]
Hello, 

as watching `storeToRefs `variables, the watch function triggers  twice, and  as the Pinia documentation states for this case we should use $subscribe.

Is there a way to subscribe to a specific mutation or state change?

This problem with the solution belove is that the .key property is removed in the prod bundle.
```
dashboardStore.$subscribe((mutation, state) => {
  console.log(mutation,state)
  if (mutation.events.key === "dashboardLoacked") {
    //do that one thing
  }
})
```

**Comment by posva** [comment, marked as answer]
You can just watch the property directly:

```ts
watch(() => dashboardStore.dashboardLoacked, ...)
```

**Reply by T0miii** [reply]
we did try this, and this watch will also trigger twice just as if we would watch storeToRefs variables.

**Reply by posva** [reply]
when does it trigger twice? what are you doing?

**Reply by dreambo8563** [reply]
@T0miii I just test it locally, the watch works (just once), provide a simple demo will helpful to find the reason

**Reply by T0miii** [reply]
ill try to provide an example online

**Reply by alizadeh118** [reply]
It works fine but what about the case we need `detached: true`? I need to watch a specific key of store and also I want my watcher to run even my component detached. any suggestion?

**Comment by Ferym26** [comment]
I still don't understand how to subscribe on specific state changes((

`export const useCounterStore = defineStore("counter", {
	state: () => ({
		count: 8,
		arr: ['item1'],
	}),
	actions: {
		increment() {
			this.count++;
		},
		addToArr() {
			setTimeout(() => {
				this.arr.push('item2')
			}, 3000)
		},
	},
});`

how to get callback on arr changes only? not count

**Comment by T0miii** [comment]
In yout component where you want to react to the change

`watch(arr, () => { console.log('the arr changed!')}, {deep:true})`

**Comment by DespertaWeb** [comment]
It happens the same to me, in the Docs is not clear how to subscribe to state in the Store. Exmple:
```
state: {
    books: [],
    category: null
 },

actions: {
    getBooks()
},
```
I'd like to subscribe to category so it triggers the getBooks() actions anytime category is changed in the store so all the logic is bundled  no in the component but in the store

**Reply by dreambo8563** [reply]
see https://github.com/vuejs/pinia/discussions/794
